tests/phpunit/api/v3/JobProcessMailingTest.php

```diff
@@ -108,6 +108,34 @@ public function testBasic() {
     $this->_mut->assertRecipients($this->getRecipients(1, 2));
   }
 
+  /**
+   * Test pause and resume on Mailing.
+   */
+  public function testPauseAndResumeMailing() {
+    $this->createContactsInGroup(10, $this->_groupID);
+    Civi::settings()->add(array(
+      'mailerBatchLimit' => 2,
+    ));
+    //Create a test mailing and check if the status is set to Scheduled.
+    $result = $this->callAPISuccess('mailing', 'create', $this->_params);
+    $jobs = $this->callAPISuccess('mailing_job', 'get', array('mailing_id' => $result['id']));
+    $this->assertEquals('Scheduled', $jobs['values'][$jobs['id']]['status']);
+
+    CRM_Mailing_BAO_MailingJob::pause($result['id']);
+    $jobs = $this->callAPISuccess('mailing_job', 'get', array('mailing_id' => $result['id']));
+    $this->assertEquals('Paused', $jobs['values'][$jobs['id']]['status']);
+
+    //Verify if Paused mailing isn't considered in process_mailing job.
+    $this->callAPISuccess('job', 'process_mailing', array());
```

Comment:
Hooray for adding test-coverage with new functionality. 🥇 

This test seems to be mostly geared toward checking that the `civicrm_mailing_job.status` changes.

IMHO, it'd be more interesting to test the behavior -- e.g. *when it's paused, does it send messages or not*?

Perhaps you could incorporate a few calls to `assertRecipients()`, as in [testBasic()](https://github.com/jitendrapurohit/civicrm-core/blob/4aa8f8046cbf703c104feba06a0b0c3d621ddf4c/tests/phpunit/api/v3/JobProcessMailingTest.php#L100-L108)?

Reply:
Done 👍